codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -69,60 +69,113 @@
     private final Http2Settings initialSettings;
     private ChannelFutureListener closeListener;
     private BaseDecoder byteDecoder;
-    private long gracefulShutdownTimeoutMillis = DEFAULT_GRACEFUL_SHUTDOWN_TIMEOUT_MILLIS;
+    private long gracefulShutdownTimeoutMillis;
 
-    public Http2ConnectionHandler(boolean server) {
-        this(server, true);
+    public static final class Builder extends BuilderBase<Http2ConnectionHandler, Builder> {
```

Comment:
I think I prefer to keep the builder in this class. It is effectively part of this class and should have access to restricted constructors.

Having `BuilderBase` and a concrete externally visible `Builder` implementation allows the implementation details of a polymorphic builder to be hidden from the user, but still allow for sub classes to inherit and extend the common builder functionality.  What did you have in mind if these two classes were to be consolidated?